terraform/monitoring/03_service_slo.tf

```diff
@@ -12,31 +12,92 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Define specifics for each of the services that will receive SLOs through a custom service
+# The data members needed are to specify the custom service, metrics for SLOs, and goals for each SLO
+variable "custom_services" {
+  type = list(object({
+    service_name = string,
+    service_id = string,
+    availability_goal = number,
+    availability_burn_rate = number,
+    latency_goal = number,
+    latency_threshold = number,
+    latency_burn_rate = number
+  }))
+  default = [
+    {
+      service_name = "Frontend Service"
+      service_id = "frontend"
+      availability_goal = 0.9
+      availability_burn_rate = 2
+      latency_goal = 0.9
+      latency_threshold = 500
+      latency_burn_rate = 2
+    },
+    {
+      service_name = "Checkout Service"
+      service_id = "checkoutservice"
+      availability_goal = 0.99
+      availability_burn_rate = 2
+      latency_goal = 0.99
+      latency_threshold = 500
+      latency_burn_rate = 2
+    },
+    {
+      service_name = "Payment Service"
+      service_id = "paymentservice"
+      availability_goal = 0.99
+      availability_burn_rate = 2
+      latency_goal = 0.99
+      latency_threshold = 500
+      latency_burn_rate = 2
+    },
+    {
+      service_name = "Email Service"
+      service_id = "emailservice"
+      availability_goal = 0.99
+      availability_burn_rate = 2
+      latency_goal = 0.99
+      latency_threshold = 500
+      latency_burn_rate = 2
+    },
+    {
+      service_name = "Shipping Service"
+      service_id = "shippingservice"
+      availability_goal = 0.99
+      availability_burn_rate = 2
+      latency_goal = 0.99
+      latency_threshold = 500
+      latency_burn_rate = 2
+    }
+  ]
+}
+
```

Comment:
I noticed that a lot of these numbers for `latency_threshold` or `availability_burn_rate` are the same across services. Would it make sense to define constants and use that for each of the services?

Reply:
Yeah the numbers are pretty similar - It's definitely possible to use a constant for them but it would require data sources in terraform which would clutter the config. I think also keeping the values like this allows us to change the values later and shows customers how to choose different values for each service.

If there is strong support towards consolidating towards constants then please let me know!